extensions-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessAction.scala

```diff
@@ -68,44 +69,34 @@ object ProcessActionState extends Enumeration {
   val FinishedStates: Set[ProcessActionState] = Set(Finished, ExecutionFinished)
 }
 
-final case class ScenarioActionName(value: String) extends AnyVal {
+sealed abstract class ScenarioActionName(override val entryName: String) extends EnumEntry {
+  def value: String             = entryName
   override def toString: String = value
 }
 
-object ScenarioActionName {
+object ScenarioActionName extends Enum[ScenarioActionName] {
 
-  implicit val encoder: Encoder[ScenarioActionName] = deriveUnwrappedEncoder
-  implicit val decoder: Decoder[ScenarioActionName] = deriveUnwrappedDecoder
+  implicit val encoder: Encoder[ScenarioActionName] = Encoder.encodeString.contramap(_.value)
+  implicit val decoder: Decoder[ScenarioActionName] =
+    Decoder.decodeString.emap(ScenarioActionName.withNameEither(_).left.map(_.getMessage()))
 
   implicit val keyEncoder: KeyEncoder[ScenarioActionName] = KeyEncoder.encodeKeyString.contramap(_.value)
-  implicit val keyDecoder: KeyDecoder[ScenarioActionName] = KeyDecoder.decodeKeyString.map(ScenarioActionName(_))
-
-  val Deploy: ScenarioActionName    = ScenarioActionName("DEPLOY")
-  val Redeploy: ScenarioActionName  = ScenarioActionName("REDEPLOY")
-  val Cancel: ScenarioActionName    = ScenarioActionName("CANCEL")
-  val Archive: ScenarioActionName   = ScenarioActionName("ARCHIVE")
-  val UnArchive: ScenarioActionName = ScenarioActionName("UNARCHIVE")
-  // TODO remove unused action
-  val Pause: ScenarioActionName  = ScenarioActionName("PAUSE") // TODO: To implement in future..
-  val Rename: ScenarioActionName = ScenarioActionName("RENAME")
-  // TODO: We kept the old name of "run now" CustomAction for compatibility reasons.
-  //       In the future it can be changed to better name, according to convention, but that would require database migration
-  //       In the meantime, there are methods serialize and deserialize, which operate on name RUN_OFF_SCHEDULE instead.
-  val RunOffSchedule: ScenarioActionName = ScenarioActionName("run now")
+  implicit val keyDecoder: KeyDecoder[ScenarioActionName] = KeyDecoder.decodeKeyString.map(ScenarioActionName.withName)
+
+  case object Deploy         extends ScenarioActionName("DEPLOY")
+  case object Redeploy       extends ScenarioActionName("REDEPLOY")
+  case object Cancel         extends ScenarioActionName("CANCEL")
+  case object Archive        extends ScenarioActionName("ARCHIVE")
+  case object UnArchive      extends ScenarioActionName("UNARCHIVE")
+  case object Pause          extends ScenarioActionName("PAUSE") // TODO: remove unused action or implement in future..
```

Comment:
Hmm, let's remove it if we don't use it..

Reply:
We do not support it, but the logic for Pause is already implemented in multiple places. Perhaps it could be removed, but I would prefer to do it outside of this change